executor/builder.go

```diff
@@ -1809,7 +1809,13 @@ func (b *executorBuilder) buildIndexLookUpJoin(v *plannercore.PhysicalIndexJoin)
 	if v.KeepOuterOrder {
 		return e
 	}
-	return &IndexNestedLoopHashJoin{IndexLookUpJoin: *e}
+	idxHash := &IndexNestedLoopHashJoin{IndexLookUpJoin: *e}
```

Comment:
Maybe we can move these code block to `newInnerWorker`.

Reply:
seems `newJoiner` rely on properties of physical plan,  I tried to put `joiners` initilized with `joinChkResourceCh`  in `startWorkers` function at start